corehq/motech/repeaters/static/repeaters/js/bootstrap3/repeat_record_report.js

```diff
@@ -292,5 +290,20 @@ hqDefine('repeaters/js/bootstrap3/repeat_record_report', function () {
         function getFlag() {
             return $confirmButton.attr('data-flag');
         }
+
+        function showWarning(reason) {
```

Comment:
I don't think this is better than the previous code, but I don't think it's worse either.

Pro: the calling code is more concise
Con: magic string as a parameter

Reply:
Yeah, that's fair. Do you think it would be nicer at all to pass in the element id we're going to show? At least then it's not pulled out of nowhere. Imagining that could look something like:
```javascript
        function showWarning(elementToShow) {
            const warningElements = ['#no-selection', '#not-allowed'];
            const elementToHide = warningElements.find(el => el !== elementToShow);
            $(elementToShow).removeClass('hide');
            $(elementToHide).addClass('hide');
        }
```

Reply:
hmm...maybe a little bit? I think the nicer thing for the calling code would be a `showWarning` and a `hideWarning`, so it doesn't need to know anything about element names. But then you'll likely end up with three functions (I'm imagining both of those would call a shared helper, which is kind of a lot for a piece of functionality that's this small.